cloud-sql/mysql/mysql/server.js

```diff
@@ -86,18 +86,34 @@ const createPool = async () => {
     //[END_EXCLUDE]
   });
 };
-createPool();
 // [END cloud_sql_mysql_mysql_create]
 
 const ensureSchema = async () => {
   // Wait for tables to be created (if they don't already exist).
-  await pool.query(
-    `CREATE TABLE IF NOT EXISTS votes
-      ( vote_id SERIAL NOT NULL, time_cast timestamp NOT NULL,
-      candidate CHAR(6) NOT NULL, PRIMARY KEY (vote_id) );`
-  );
+  try {
+    await pool.query(
+      `CREATE TABLE IF NOT EXISTS votes
+        ( vote_id SERIAL NOT NULL, time_cast timestamp NOT NULL,
+        candidate CHAR(6) NOT NULL, PRIMARY KEY (vote_id) );`
+    );
+    console.log(`Ensured that table 'votes' exists`);
+  } catch (e) {
+    console.log(`Got error: ${e}`);
+  }
 };
-ensureSchema();
+
+let schemaReady;
+
+createPool()
+  .then(() => (schemaReady = ensureSchema()))
+  .catch((error) => console.log(error));
+
+const awaitSchema = async (req, res, next) => {
+  await schemaReady;
+  next();
+};
+
+app.use(awaitSchema);
```

Comment:
Why is there `awaitSchema` and `ensureSchema`? I'm not sure what `next()` is doing. Is it possible to put them in a single function?

Reply:
3rd parameter tells the express app to wait for the middlewear to finish rather than to start them all at once. Looks like perfectly valid use of that pattern here.

Reply:
Thanks for the explanation!